File: DATN_GameVH/src/main/java/com/gamevh/service/OrderService.java
package com.gamevh.service;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

import com.gamevh.dto.FullOrderDTO;
import com.gamevh.entities.OrderData;
import com.gamevh.handle.CustomException;
import org.springframework.data.domain.Page;


public interface OrderService {
	
	List<OrderData> findAll();
	Page<OrderData> findAll(Optional<Integer> page, Optional<Integer> size, String username, String phone, Optional<LocalDate> createdAt);

	OrderData findById(Long id);

	OrderData findByOrderId(String orderId);
	
	OrderData add(OrderData orderData);
	FullOrderDTO findOne(String orderId) throws CustomException;
	void updateStatus(String orderId, String newStatus) throws CustomException;
}

File: GamesVH-main/DATN_GameVH_FE/admin/services/api/OrderService.js
app.factory("OrderService", function ($http) {
  const baseUrl = host + "/api/orderdata";
  return {
    getAll: function (filters) {
      const queries = {
        page: filters.page || 0,
        size: filters.size || 500,
        username: filters.username || null,
        phone: filters.phone || null,
        createdAt: filters.createdAt || null,
      };
      console.log(queries);
      return $http.get(baseUrl, { params: queries });
    },
    getOne(id) {
      return $http.get(`${baseUrl}/full/${id}`);
    },
    getTop5() {
      return $http.get(baseUrl + "/getTop5");
    },
    updateStatus(orderId, newStatus) {
      return $http.put(`${baseUrl}/status/${orderId}`, newStatus);
    },
  };
});
